docs/USING_PUBLIC_CLOUDS.rst

```diff
@@ -0,0 +1,459 @@
+   -v $(pwd)/locationConfig.json:/usr/src/app/locationConfig.json \
+   -v $(pwd)/conf/authdata.json:/usr/src/app/conf/authdata.json \
+   -v ~/.aws/credentials:/root/.aws/credentials \
+   -e S3DATA=multiple -e ENDPOINT=http://localhost -p 8000:8000
+   -d scality/s3server
+
+Testing: put an object to AWS S3 using CloudServer
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In order to start testing pushing to AWS S3, you will need to create a local
+bucket in the AWS S3 location constraint - this local bucket will only store the
+metadata locally, while both the data and the metadata will be stored on AWS S3.
+This example is based on all our previous steps.
+
+.. code:: shell
+
+   # Create a local bucket hosting data in AWS S3
+   $> s3cmd --host=127.0.0.1:8000 mb s3://zenkobucket --region=aws-test
+   # Put an object to AWS S3, and store the metadata locally
+   $> s3cmd --host=127.0.0.1:8000 put /etc/hosts s3://zenkobucket/testput
+    upload: '/etc/hosts' -> 's3://zenkobucket/testput'  [1 of 1]
+     330 of 330   100% in    0s   380.87 B/s  done
+   # List locally to check you have the metadata
+   $> s3cmd --host=127.0.0.1:8000 ls s3://zenkobucket
+    2017-10-23 10:26       330   s3://zenkobucket/testput
+
+Then, from the AWS Console, if you go into your bucket, you should see your
+newly uploaded object:
+
+.. figure:: ../res/aws-console-successful-put.png
+   :alt: AWS S3 Console upload example
+
+Troubleshooting
+~~~~~~~~~~~~~~~
+
+Make sure your :code:`~/.s3cfg` file has credentials matching your local
+CloudServer credentials defined in :code:`conf/authdata.json`. By default, the
+access key is :code:`accessKey1` and the secret key is :code:`verySecretKey1`.
+For more informations, refer to our template `~/.s3cfg <./CLIENTS/#s3cmd>`__ .
+
+Pre existing objects in your AWS S3 hosted bucket can unfortunately not be
+accessed by CloudServer at this time.
+
+Make sure versioning is enabled in your remote AWS S3 hosted bucket. To check,
+using the AWS Console, click on your bucket name, then on "Properties" at the
+top, and then you should see something like this:
+
+.. figure:: ../res/aws-console-versioning-enabled.png
+   :alt: AWS Console showing versioning enabled
+
+Microsoft Azure as a backend
+----------------------------
+
+From the MS Azure Console
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+From your Stoage Account dashboard, create a container where you will host your
+data for this new location constraint.
+
+You will also need to generate a SAS (Shared Access Signature) from the Console.
+This can be done from your Storage Account dashboard, under "Settings, then
+"Shared access signature". Your SAS will expire after some time. You can give it
+different permissions.
+Once you click "Generate SAS", a "SAS token" and a "Blob service SAS URL" will
+appear. You want to use the **SAS token without the initial question mark**.
+
+In this example, our container will be name ``zenkontainer``, and will belong to
```

Comment:
will be called

Reply:
changed to "named" for consistency with your earlier comment on AWS S3 as a data backend